src/test/java/it/eng/idsa/dataapp/handler/ContractRequestMessageHandlerTest.java

```diff
@@ -55,10 +65,35 @@ void handleMessageTest() {
 		assertTrue(StringUtils.containsIgnoreCase(responseMap.get("header").toString(), message.getId().toString()));
 	}
 
+	@Test
+	void handleMessageWithoutContractNegotiationExceptionTest() {
+		ReflectionTestUtils.setField(contractRequestMessageHandler, "contractNegotiationDemo", false);
+
+		InternalRecipientException exception = assertThrows(InternalRecipientException.class, () -> {
+			responseMap = contractRequestMessageHandler.handleMessage(message, "asdasdasd");
+		});
+
+		assertEquals("Creating processed notification, contract agreement needs evaluation", exception.getMessage());
+	}
+
+	@Test
+	void handleMessagePlatoonNotFoundExceptionTest() {
+
+		ReflectionTestUtils.setField(contractRequestMessageHandler, "usageControlVersion", usageControlVersionPlatoon);
+		when(selfDescriptionService.getSelfDescription(message)).thenReturn(baseConnector);
+		NotFoundException exception = assertThrows(NotFoundException.class, () -> {
+			responseMap = contractRequestMessageHandler.handleMessage(message,
+					"{\"ids:consumer\":{\"@id\":\"http://w3id.org/engrd/connector/consumer\"},\"@type\":\"ids:ContractRequest\",\"ids:provider\":{\"@id\":\"https://w3id.org/engrd/connector/\"},\"ids:permission\":[{\"ids:postDuty\":[],\"ids:action\":[{\"@id\":\"https://w3id.org/idsa/code/USE\"}],\"ids:constraint\":[],\"ids:assignee\":[],\"@type\":\"ids:Permission\",\"ids:title\":[{\"@value\":\"Example Usage Policy\",\"@type\":\"http://www.w3.org/2001/XMLSchema#string\"}],\"ids:preDuty\":[],\"ids:description\":[{\"@value\":\"provide-access\",\"@type\":\"http://www.w3.org/2001/XMLSchema#string\"}],\"ids:target\":{\"@id\":\"http://w3id.org/engrd/connector/artifact/1\"},\"@id\":\"https://w3id.org/idsa/autogen/permission/5c5b8374-2597-4c5e-b694-494ccaa68136\",\"ids:assigner\":[]}],\"@id\":\"https://w3id.org/idsa/autogen/contractOffer/008c7957-1e12-4538-9ba0-22d49216e578\",\"ids:prohibition\":[],\"ids:obligation\":[],\"@context\":{\"ids\":\"https://w3id.org/idsa/core/\",\"idsc\":\"https://w3id.org/idsa/code/\"}}");
```

Comment:
Can you use creating contract from test classes (in multipart) and convert that to string and use it like that, instead this "magic string"?